packages/backend/src/apps/gathersg/auth/decrypt-response.ts

```diff
@@ -0,0 +1,106 @@
+import { IGlobalVariable } from '@plumber/types'
+
+import crypto from 'crypto'
+
+import appConfig from '@/config/app'
+import logger from '@/helpers/logger'
+
+import schema from './schema'
+
+function getInternalId(data: any) {
+  let internalId
+  if ('updatedAt' in data) {
+    // try to find the updatedAt first
+    internalId = `${data?.uuid}-${data.updatedAt}`
+  } else if ('createdAt' in data) {
+    // otherwise fallback to createdAt, for newly created cases
+    internalId = `${data?.uuid}-${data.createdAt}`
+  }
+  return internalId
```

Comment:
nit: `in` operator will error if data is undefined
```
function getInternalId(data: any) {
  if (data?.updatedAt) {
    // try to find the updatedAt first
    return `${data?.uuid}-${data.updatedAt}`
  }
  if (data?.createdAt) {
    // otherwise fallback to createdAt, for newly created cases
    return `${data?.uuid}-${data.createdAt}`
  }
  return ''
}
```